core/chains/evm/client/client.go

```diff
@@ -197,6 +202,107 @@ func (client *client) HeaderByHash(ctx context.Context, h common.Hash) (*types.H
 	return client.pool.HeaderByHash(ctx, h)
 }
 
+func (client *client) SendTransactionAndReturnErrorType(ctx context.Context, tx *types.Transaction, fromAddress common.Address) (txmtypes.TxmErrorType, error) {
+	err := client.pool.SendTransaction(ctx, tx)
+	sendError := NewSendError(err)
+	if sendError.Fatal() {
+		client.logger.Criticalw("Fatal error sending transaction", "err", sendError, "etx", tx)
+		// Attempt is thrown away in this case; we don't need it since it never got accepted by a node
+		return txmtypes.Fatal, err
+	}
+	if sendError.IsNonceTooLowError() || sendError.IsTransactionAlreadyMined() {
```

Comment:
IsTransactionAlreadyMined() case should be returning Successful, similar to IsTransactionAlreadyInMempool().